provider/ollama/ollama.go

```diff
@@ -0,0 +1,125 @@
+package ollama
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/sashabaranov/go-openai"
+
+	"github.com/symflower/eval-dev-quality/log"
+	"github.com/symflower/eval-dev-quality/model"
+	"github.com/symflower/eval-dev-quality/model/llm"
+	"github.com/symflower/eval-dev-quality/provider"
+	"github.com/symflower/eval-dev-quality/tools"
+)
+
+// Provider holds an "Ollama" provider.
+type Provider struct {
+	// binaryPath holds the binary file path of the Ollama binary.
+	binaryPath string
+	// url holds the URL to the Ollama service.
+	url string
+}
+
+func init() {
+	provider.Register(NewProvider())
+}
+
+// NewProvider returns an "Ollama" provider.
+func NewProvider() (provider provider.Provider) {
+	return &Provider{
+		binaryPath: tools.OllamaPath,
+		url:        tools.OllamaURL,
+	}
+}
+
+// SetBinaryPath sets the binary file path of the Ollama binary.
+func (p *Provider) SetBinaryPath(binaryPath string) {
+	p.binaryPath = binaryPath
+}
+
+// SetURL sets the URL to the Ollama service.
+func (p *Provider) SetURL(url string) {
+	p.url = url
+}
```

Comment:
Only used in tests, so I'd say could also directly modify the struct content and get rid of this function.